src/dcrypto/dcrypto_test.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dcrypto
+
+import (
+	"bytes"
+	"crypto/rand"
+	"io/ioutil"
+	"testing"
+)
+
+var password = []byte("test")
+
+// randBytes returns random bytes in a byte slice of size.
+func randBytes(size int) ([]byte, error) {
+	b := make([]byte, size)
+	_, err := rand.Read(b)
+	return b, err
+}
+
+// TestRoundTrip tests several size sets of data going through the encrypt/decrypt
+// to make sure they come out the same.
+func TestRoundTrip(t *testing.T) {
+	for _, size := range []int{24, 66560} {
+		t.Logf("Testing file of size: %db", size)
+		f, err := randBytes(size)
+		if err != nil {
+			t.Errorf("randBytes(%d) => %q; want nil", size, err)
+			continue
+		}
+		er, err := NewEncrypter(bytes.NewBuffer(f), password)
+		if err != nil {
+			t.Errorf("NewEncrypter() => %q; want nil", err)
+			continue
+		}
+		cipher, err := ioutil.ReadAll(er)
+		if err != nil {
+			t.Errorf("ioutil.ReadAll(*Encrypter) => %q; want nil", err)
+			continue
+		}
+		dr, err := NewDecrypter(bytes.NewBuffer(cipher), password)
```

Comment:
I can do this but each test case takes ~2s to run since the key generation has to take place twice. Adding 0 and an odd test case has us at 8s so each password would add another 8s. I'm fine adding more just wanted you to be aware.